### ISSUE TITLE
cordova-android@12 app crashing and not install issu

### ISSUE THREAD

**Issue by hansarajgyaneswar** [apache/cordova-android#1619]
![IMG-20230528-WA0010](https://github.com/apache/cordova-android/assets/103986472/8dba1540-ee59-4dd4-9b13-8c9a6b4b41bc)
![IMG-20230528-WA0009](https://github.com/apache/cordova-android/assets/103986472/7f9bd582-67f5-4db1-80fb-c63113fc276b)
# Bug Report

## Checklist

- [ ] I searched for existing GitHub issues
- [ ] I updated all Cordova tooling to most recent version
- [ ] I included all the necessary information above

**Comment by hansarajgyaneswar** [comment]
Many devices opening splash screen and close

**Comment by breautek** [comment]
Please fill out the form, and include the output of `cordova plugin ls`

**Comment by hansarajgyaneswar** [comment]
cordova-plugin-admobpro 8.13.1 "AdMob Plugin Pro"
cordova-plugin-contacts-phonenumbers 0.0.12 "Contacts Phone Numbers"
cordova-plugin-device 2.1.0 "Device"
cordova-plugin-extension 1.6.0 "Cordova Plugin Extension"
cordova-plugin-geolocation 4.1.0 "Geolocation"
cordova-plugin-upi 1.0.4 "UPI"

**Comment by breautek** [comment]
Can you explain the steps that was taken to update to cordova-android@12?

**Comment by hansarajgyaneswar** [comment]
First unstall Cordova and fresh install

**Comment by hansarajgyaneswar** [comment]
App working fine my emulator and my mobile. But some device app crash and not opening and installing issue

**Comment by hansarajgyaneswar** [comment]
![IMG-20230529-WA0007](https://github.com/apache/cordova-android/assets/103986472/271a7c17-6ade-4b49-97ad-59e6baa870e8)
My friend device not working 

I already did above guide

**Comment by hansarajgyaneswar** [comment]
/
![Screenshot_2023-05-29-22-26-31-640_com example hello](https://github.com/apache/cordova-android/assets/103986472/5b31b78f-b563-4748-8128-8b0d71761e58)
My device working fine

**Comment by hansarajgyaneswar** [comment]
I set also minimum sdk 21

**Comment by breautek** [comment]
> I set also minimum sdk 21

cordova-android@12 is only tested with API 24+.

> ![IMG-20230529-WA0007](https://user-images.githubusercontent.com/103986472/241777314-271a7c17-6ade-4b49-97ad-59e6baa870e8.jpg) My friend device not working
> 
> I already did above guide

This looks like a native UI being produced by a plugin given that it has an action bar, which the cordova activity has disabled. The problem here appears to be in one of your installed plugins.

**Comment by hansarajgyaneswar** [comment]
I did without any plugin
I make simple hello world app  but same problem above screenshot attached 

/
![Screenshot_2023-05-29-22-26-31-640_com example hello](https://github.com/apache/cordova-android/assets/103986472/24b099b7-9e08-4d98-9c16-f9c52462d1c4)
![IMG-20230529-WA0007](https://github.com/apache/cordova-android/assets/103986472/0cc1fd29-8b9d-4ac3-87a1-92dc9d1765cc)

**Comment by hansarajgyaneswar** [comment]
FATAL EXCEPTION: main
                                                                                                    Process: com.example.hello, PID: 4800
                                                                                                    java.lang.NoClassDefFoundError: org.apache.cordova.PluginManager$$ExternalSyntheticLambda0
                                                                                                    	at org.apache.cordova.PluginManager.postMessage(PluginManager.java:355)
                                                                                                    	at org.apache.cordova.CordovaActivity.init(CordovaActivity.java:161)
                                                                                                    	at org.apache.cordova.CordovaActivity.loadUrl(CordovaActivity.java:234)
                                                                                                    	at com.example.hello.MainActivity.onCreate(MainActivity.java:40)
                                                                                                    	at android.app.Activity.performCreate(Activity.java:6237)
                                                                                                    	at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1107)
                                                                                                    	at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2369)
                                                                                                    	at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2476)
                                                                                                    	at android.app.ActivityThread.-wrap11(ActivityThread.java)
                                                                                                    	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1344)
                                                                                                    	at android.os.Handler.dispatchMessage(Handler.java:102)
                                                                                                    	at android.os.Looper.loop(Looper.java:148)
                                                                                                    	at android.app.ActivityThread.main(ActivityThread.java:5417)
                                                                                                    	at java.lang.reflect.Method.invoke(Native Method)
                                                                                                    	at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:726)
                                                                                                    	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:616)

**Comment by hansarajgyaneswar** [comment]
app not ruuning adroid 5 6 on  e'4_WVGA_Nexus_S_Edited_API_23 [emulator-5554]'.

**Comment by breautek** [comment]
> I did without any plugin I make simple hello world app but same problem above screenshot attached
> 
> / ![Screenshot_2023-05-29-22-26-31-640_com example hello](https://user-images.githubusercontent.com/103986472/241826440-24b099b7-9e08-4d98-9c16-f9c52462d1c4.jpg) ![IMG-20230529-WA0007](https://user-images.githubusercontent.com/103986472/241826446-0cc1fd29-8b9d-4ac3-87a1-92dc9d1765cc.jpg)

The first screenshot does like the cordova hello world template. The second screenshot that says "App not installed." is not something cordova produces.

> app not ruuning adroid 5 6 on e'4_WVGA_Nexus_S_Edited_API_23 [emulator-5554]'.

API 23 is not supported by the cordova framework so your mileage may vary if you try to force it (by overriding the `android-minSdkVersion` for example). On cordova-android@12, the minimum SDK is 24. For the purposes of figuring out the issue, we should be testing no earlier than API 24 to rule out the possibility that the problem is not a problem due to using an unsupported API.

> FATAL EXCEPTION: main
Process: com.example.hello, PID: 4800
java.lang.NoClassDefFoundError: org.apache.cordova.PluginManager$$ExternalSyntheticLambda0
at org.apache.cordova.PluginManager.postMessage(PluginManager.java:355)
at org.apache.cordova.CordovaActivity.init(CordovaActivity.java:161)
at org.apache.cordova.CordovaActivity.loadUrl(CordovaActivity.java:234)
at com.example.hello.MainActivity.onCreate(MainActivity.java:40)
at android.app.Activity.performCreate(Activity.java:6237)
at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1107)
at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2369)
at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2476)
at android.app.ActivityThread.-wrap11(ActivityThread.java)
at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1344)
at android.os.Handler.dispatchMessage(Handler.java:102)
at android.os.Looper.loop(Looper.java:148)
at android.app.ActivityThread.main(ActivityThread.java:5417)
at java.lang.reflect.Method.invoke(Native Method)
at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:726)
at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:616)

I've tested the hello world app with your plugins installed. API 24 works (as in it launches without issues)
API 23 produces the above stacktrace. This is because API 23 doesn't have support for lambda functions on a collection.forEach API which the framework uses. As I mentioned before, the Cordova Framework supports API 24+. (See  the [PR](https://github.com/apache/cordova-android/pull/1571) and [Mail Thread](https://lists.apache.org/thread/zcgof080hdzzo2j96mjz0qpj0gotmn57) for the rationale behind bumping to API 24).

If you truly need API 23 support, you'll have to fork the cordova framework and refactor out anything that requires API 24+.

Note that I also did some testing on API 33 devices and some of your plugins doesn't appear to support API 31+ either (they require changes in their usage of `PendingIntent`). This includes

1. cordova-plugin-admobpro which triggers a crash on startup because it needs to explicitly set a `FLAG_MUTABLE` or `FLAG_IMMUTABLE` on the `PendingIntent` it creates.
2. cordova-plugin-upi also has the same issue on the version released to NPM, but it does appear to be fixed in their main branch.

None of the issues discovered are bugs with Cordova, so I'll be closing this issue.

**Comment by hansarajgyaneswar** [comment]
Simple hello world app not installing

**Comment by hansarajgyaneswar** [comment]
On Android 9

**Comment by kosbar** [comment]
But PluginManager has this code to prevent problems with Lambda (@breautek):
```
            if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.N) {
                this.pluginMap.forEach((s, plugin) -> {
                    if (plugin != null) {
                        plugin.onMessage(id, data);
                    }
                });
            } else {
                for (CordovaPlugin plugin : this.pluginMap.values()) {
                    if (plugin != null) {
                        Object obj = plugin.onMessage(id, data);
                        if (obj != null) {
                            return obj;
                        }
                    }
                }
            }
```
We use cordova-android 11 with 
`<preference name="android-minSdkVersion" value="21" />`

**Comment by breautek** [comment]
> But PluginManager has this code to prevent problems with Lambda (@breautek):
> 
> ```
>             if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.N) {
>                 this.pluginMap.forEach((s, plugin) -> {
>                     if (plugin != null) {
>                         plugin.onMessage(id, data);
>                     }
>                 });
>             } else {
>                 for (CordovaPlugin plugin : this.pluginMap.values()) {
>                     if (plugin != null) {
>                         Object obj = plugin.onMessage(id, data);
>                         if (obj != null) {
>                             return obj;
>                         }
>                     }
>                 }
>             }
> ```
> 
> We use cordova-android 11 with `<preference name="android-minSdkVersion" value="21" />` Some time it works on 21 API (debug version from AS) sometimes crashes (cordova build debug with signature)...

cordova-android@12 has the check removed to simplify the codebase so it only has the lambda usage. Cordova-android@11 has the SDK version check to support both.

This is getting off-topic now so I'm locking the thread.

TL;DR; is don't use a minimum SDK less than 24 on cordova-android@12.
